src/ride/ride_ratings.c

```diff
@@ -599,17 +599,214 @@ static int sub_65E72D(rct_ride *ride)
 	return edx & 0xFFFF;
 }
 
+static rating_tuple get_var_10E_rating(uint16 var_10E) {
+	int al;
+	al = var_10E >> 8;
+	al = al & 7;
+	int excitement = (al * 163840) >> 16;
+
+	al = var_10E >> 5;
+	al = al & 7;
+	excitement += (al * 196608) >> 16;
+
+	// Low bits, 0x0065DF3A
+	al = var_10E & 0x1F;
+	excitement += (al * 63421) >> 16;
+
+	al = var_10E >> 8;
+	al = al & 7;
+	int intensity = (al * 81920) >> 16;
+
+	al = var_10E >> 5;
+	al = al & 7;
+	intensity += (al * 49152) >> 16;
+
+	al = var_10E & 0x1F;
+	intensity += (al * 21140) >> 16;
+
+	al = var_10E >> 8;
+	al = al & 7;
+	int nausea = (al * 327680) >> 16;
+
+	al = var_10E >> 5;
+	al = al & 7;
+	nausea += (al * 204800) >> 16;
+
+	al = var_10E & 0x1F;
+	nausea += (al * 42281) >> 16;
+
+	rating_tuple rating = { excitement, intensity, nausea };
+	return rating;
+}
+
+/**
+ * rct2: 0x0065DF72
+ */
+static rating_tuple get_var_110_rating(uint16 var_110) {
+	int al;
+
+	al = var_110 >> 8;
+	al = al & 7;
+	int excitement = (al * 245760) >> 16;
+
+	al = var_110 >> 5;
+	al = al & 7;
+	excitement += (al * 245760) >> 16;
+
+	al = var_110 & 0x1F;
+	excitement += (al * 73992) >> 16;
+
+	al = var_110 >> 8;
+	al = al & 7;
+	int intensity = (al * 81920) >> 16;
+
+	al = var_110 >> 5;
+	al = al & 7;
+	intensity += (al * 49152) >> 16;
+
+	al = var_110 & 0x1F;
+	intensity += (al * 21140) >> 16;
+
+	al = var_110 >> 8;
+	al = al & 7;
+	int nausea = (al * 327680) >> 16;
+
+	al = var_110 >> 5;
+	al = al & 7;
+	nausea += (al * 204800) >> 16;
+
+	al = var_110 & 0x1F;
+	nausea += (al * 48623) >> 16;
+
+	rating_tuple rating = { excitement, intensity, nausea };
+	return rating;
+}
+
 /**
+ * rct2: 0x0065E047
+ */
+static rating_tuple get_var_112_rating(uint16 var_112) {
+	int al;
+
+	al = var_112 >> 11;
+	al = min(al & 0x3F, 4);
+	int excitement = (al * 491520) >> 16;
+
+	al = var_112 >> 11;
+	al = min(al & 0x3F, 8);
+	int nausea = (al * 491520) >> 16;
+
+	al = var_112 >> 8;
+	al = min(al & 7, 6);
+	excitement += (al * 273066) >> 16;
+
+	al = var_112 >> 5;
+	al = min(al & 7, 6);
+	excitement += (al * 240298) >> 16;
+
+	al = min(var_112 & 0x1F, 7);
+	excitement += (al * 187245) >> 16;
+
+	rating_tuple rating = { excitement, 0, nausea };
+	return rating;
+}
+
+/**
+ * rct2: 0x0065E0F2
+ */
+static rating_tuple get_inversions_ratings(uint8 inversions) {
+	inversions = inversions & 0x1F;
+
+	int a = min(inversions, 6);
+	int excitement = (a * 1747626) >> 16;
+
+	int intensity = (inversions * 3276800) >> 16;
+	int nausea = (inversions * 1419946) >> 16;
+
+	rating_tuple rating = { excitement, intensity, nausea };
+	return rating;
+}
+
+/*
  *
+ */
+static rating_tuple get_var_0D5_rating(uint8 type, uint8 var_0D5) {
+	int excitement = 0, intensity = 0, nausea = 0;
+	if (type == RIDE_TYPE_GHOST_TRAIN) {
+		if (var_0D5 & 0x20) {
+			excitement += 40;
+			intensity  += 25;
+			nausea     += 55;
+		}
+	} else if (type == RIDE_TYPE_LOG_FLUME) {
+		if (var_0D5 & 0x40) {
```

Comment:
These are funny because they seem to be checking if a bit is set but later var_0D5 gets subtracted by 5. Maybe the low 5 bits mean something?